### PR TITLE
Fix manual queries losing parameters when called

### LINKED ISSUE #2019: Parameter bindings are lost when manually calling a query
When defining a query that is to be called manually, one can bind the parameters via UI, like so:

However, these are required to be re-written when calling this query, for instance in a Button `onClick` binding

The expectation is that I need not have to pass in the parameters in the code, and simply write `<query-name>.call()`

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes #2019 
- Visual demonstration 
### Before

https://github.com/mui/mui-toolpad/assets/19550456/33d7e121-e372-4ab5-8b01-c33096ac8e1c

### After

https://github.com/mui/mui-toolpad/assets/19550456/89d0a8ea-1331-4ff3-8b81-cb3d46528f05

- Please follow the [the contributing guide on how to create great pull requests](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/toolpad-app/src/runtime/ToolpadApp.tsx [hunk 1]
```diff
@@ -577,6 +577,27 @@ function flattenNestedBindables(
   );
 }
 
+/**
```
Comment (apedroferreira) [inline packages/toolpad-app/src/runtime/ToolpadApp.tsx:580]: If this method is just for internal usage do we really need these annotations?
I feel like the Typescript types could be explanatory enough.
But of course it also doesn't hurt to have this, besides it just taking a few more lines.

**Review by apedroferreira — APPROVED** [review]

**Comment by Janpot** [comment]
Can you add a test case to either https://github.com/mui/mui-toolpad/blob/master/test/integration/backend-basic/index.spec.ts o  https://github.com/mui/mui-toolpad/blob/master/test/integration/rest-basic/index.spec.ts ?
Just a button that triggers a query with bound parameters and displaying the output

File: test/integration/backend-basic/fixture/toolpad/resources/functions.ts [hunk 2]
```diff
@@ -41,3 +41,15 @@ export const withParams = createFunction(
...
 );
+
+export const manualQueryWithParams = createFunction(
+  async ({ parameters }) => {
+    return { parameters };
+  },
+  {
+    parameters: {
+      foo: { type: 'string' },
+      // __NEW_PARAMETER__
```
Comment (Janpot) [inline test/integration/backend-basic/fixture/toolpad/resources/functions.ts:52]:
```suggestion
```

File: test/integration/backend-basic/fixture/toolpad/pages/page1/page.yml [hunk 3]
```diff
@@ -93,3 +120,13 @@ spec:
...
+    - name: manualQuery
+      mode: mutation
+      query:
+        function: manualQueryWithParams
+        kind: local
+      parameters:
+        - name: foo
+          value:
+            $$jsExpression: |
+              text.value.split(" ")?.[1]
```
Comment (Janpot) [inline test/integration/backend-basic/fixture/toolpad/pages/page1/page.yml:132]: Let's keep the binding simple? Any js expression will do.

```suggestion
              'checksum'
```

**Review by Janpot — APPROVED** [review]
Looking good! I'd scope down the fixture a bit to focus on passing the binding only. Complexer binding setups we already test in another integration test.

Approving already

**Pull request merged**